scripts/deploy.js

```diff
@@ -0,0 +1,22 @@
+const { ethers } = require("hardhat");
+
+async function main() {
+  console.log("signer", (await ethers.getSigner()).address)
+  console.log("signer's balance", await (await ethers.getSigner()).getBalance())
+
+  const Contract = await ethers.getContractFactory('SystemDeployer');
+  const contract = await Contract.deploy();
+  await contract.deployed();
+  console.log('Contract deployed at ' + contract.address);
+  const addr = "0x0000000000000000000000000000000003330001"
```

Comment:
The string way is vulnerable. Especially many js scripts rely on this.
Perhaps having a common config file, and forcing everyone read from it, might be a better idea?